### DISCUSSION TITLE
TensorFlowV2Classifier and SaliencyMapMethod

### DISCUSSION THREAD

**Discussion by fatimah-aloraini** [Trusted-AI/adversarial-robustness-toolbox#1928] [Q&A]
Hello, thank you for ART. It is a great help!
I am working on adversarial attacks against IDS project (binary classification: normal and malicious samples); I faced "slice index 1 of dimension 1 out of bounds. [Op:StridedSlice] name: strided_slice/" error, and the traceback is attached. I did not understand why the line:  
prediction=predictions[:unique_label] 
in TensorFlowV2Classifier causes an error

`
from art.estimators.classification import TensorFlowV2Classifier
from art.attacks.evasion import SaliencyMapMethod
classifier = TensorFlowV2Classifier(
    model=model,
    loss_object=tf.keras.losses.BinaryCrossentropy(),
    nb_classes=2,
    input_shape=x_train[0].shape,
)

JSMA= SaliencyMapMethod(classifier=classifier, theta=0.1, gamma=0.5 )
x_test_adv = JSMA.generate(np.array(x_test))`

<img width="679" alt="Trace_back_1" src="https://user-images.githubusercontent.com/118727492/203058343-163e159c-25eb-4804-96e1-e341581b375e.png">

<img width="673" alt="Trace_back_2" src="https://user-images.githubusercontent.com/118727492/203058396-a762162f-7961-4381-907a-260a0d0cd4b1.png">

**Comment by beat-buesser** [comment]
Hi @fatimah-aloraini What is the shape of the model's pedictions? What is the accuracy of the model on its training data?

**Reply by fatimah-aloraini** [reply]
Hello @beat-buesser, appreciate your response.
The shape of the model prediction is **TensorShape([219397, 1])** where 219397 is the number of samples in the testing set.  The accuracy of the model on its training set is **96%**

**Reply by beat-buesser** [reply]
Hi @fatimah-aloraini The attack uses class gradients which supports binary classification but requires a model with 2 output logits (1 for each class). Can you try with a model that has output shape TensorShape([219397, 2])?
